WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt

```diff
@@ -1139,7 +1147,15 @@ class MySiteViewModel @Inject constructor(
     }
 
     private fun selectDefaultTabIfNeeded() {
-        if (!isMySiteTabsEnabled || isDefaultABExperimentTabSet) return
+        if (!isMySiteTabsEnabled) return
+        // This logic checks if the current default tab is the same as the tab
+        // set as initial screen, if yes then return
+        if (isDefaultABExperimentTabSet) {
+            _selectTab.value?.let { tab ->
+                val currentDefaultTab = tab.peekContent().position
+                if (currentDefaultTab == orderedTabTypes.indexOf(defaultABExperimentTab)) return
+            }
+        }
         val index = orderedTabTypes.indexOf(defaultABExperimentTab)
         if (index != -1) {
```

Comment:
Changes look good to me. Also I like that initial screen when changed from App Settings selects corresponding tab without restarting the app. 👏 One minor suggestion is to put common `defaultABExperimentTab` index logic at the top:  

```suggestion
        val index = orderedTabTypes.indexOf(defaultABExperimentTab)
        if (index != -1) {
            // This logic checks if the current default tab is the same as the tab
            // set as initial screen, if yes then return
            if (isDefaultABExperimentTabSet) {
                _selectTab.value?.let { tab ->
                    val currentDefaultTab = tab.peekContent().position
                    if (currentDefaultTab == index) return
                }
            }
```

Feel free to merge after this.

Reply:
@ashiagr  🤔  . I am sorry, I didn't understand actually the reason to shift the logic below. At present, the logic of checking whether the `isDefaultABExperimentTabSet`  is before accessing the index of the `defaultExperimentTab`.  If we move the condition block with  `if (isDefaultABExperimentTabSet)` wouldn't it be accessed regardless of the value of `isDefaultABExperimentTabSet`.

Reply:
@ashiagr I have made the changes as per your suggestion in [56f442c](https://github.com/wordpress-mobile/WordPress-Android/pull/16361/commits/56f442c198739db9c0c4b4417fff084c6fd94dd6).

Reply:
LGTM, thanks, @AjeshRPai! 

:shipit: